syntax = "proto3";
package collector;





/*
DebugCollector请求
*/
message DebugCollectorRequest {
message Args {
    /*
    脚本入参值(当入参为自定义时需要)
    */
    string value = 1 ;
    /*
    脚本入参key
    */
    string key = 2 ;
    /*
    参数的来源
attr_id: CMDB模型的属性
custom: 自定义

    */
    string source = 3 ;
}
        
    
    /*
    套件id
    */
    string pluginId = 1 ;
    /*
    执行目标实例Id
    */
    string target = 2 ;
    /*
    脚本入参
    */
    repeated Args args = 3 ;
    /*
    脚本类型
    */
    string script = 4 ;
    /*
    脚本内容
    */
    string content = 5 ;
}

/*
DebugCollector返回
*/
message DebugCollectorResponse {
message MetricGroups {
message DimStatus {
    /*
    dim的状态
ok: 通过调试
undefined: 未定义
mismatched_type: 类型错误
missing: 未采集到
duplicated: 重复

如果同时具有多种状态只返回最重要的状态
1:mismatched_type和duplicated只返回mismatched_type
2:undefined和duplicated只返回undefined

    */
    string status = 1 ;
    /*
    维度id
    */
    string id = 2 ;
    /*
    维度名称
    */
    string name = 3 ;
}
        
message ValStatus {
    /*
    val的状态
ok: 通过调试
undefined: 未定义
mismatched_type: 类型错误
missing: 未采集到
duplicated: 重复

如果同时具有多种状态只返回最重要的状态
1:mismatched_type和duplicated只返回mismatched_type
2:undefined和duplicated只返回undefined

    */
    string status = 1 ;
    /*
    指标id
    */
    string id = 2 ;
    /*
    指标名称
    */
    string name = 3 ;
    /*
    指标类型
    */
    string type = 4 ;
    /*
    指标单位
    */
    string unit = 5 ;
}
        
message List {
message Dims {
    /*
    维度的值
    */
    string value = 1 ;
    /*
    维度id
    */
    string id = 2 ;
}
        
message Vals {
    /*
    val的值
    */
    string value = 1 ;
    /*
    指标id
    */
    string id = 2 ;
}
        
    
    /*
    一条数据的维度列表
    */
    repeated Dims dims = 1 ;
    /*
    一条数据的val列表
    */
    repeated Vals vals = 2 ;
}
        
    
    /*
    指标组状态
ok: 通过调试
undefined: 指标组未定义
missing: 未采集指标组
dims_not_unique: 维度不能唯一确定数据
metrics_not_ok: dims或vals不合法(未定义,类型错误,未采集到或重复)

如果同时具有多种状态只返回最重要的状态
1:metrics_not_ok和dims_not_unique只返回dims_not_unique
2:undefined和dims_not_unique只返回undefined

    */
    string metric_group_status = 1 ;
    /*
    dim的状态列表
    */
    repeated DimStatus dim_status = 2 ;
    /*
    val的状态列表
    */
    repeated ValStatus val_status = 3 ;
    /*
    指标组的值的列表
    */
    repeated List list = 4 ;
    /*
    指标组id
    */
    string id = 5 ;
    /*
    指标组名
    */
    string name = 6 ;
    /*
    两级分类，用.分割，(如：主机状态.基本配置 )
    */
    string category = 7 ;
}
        
    
    /*
    脚本执行状态
ok: 正常执行且结果通过调试,
failed: 正常执行但结果未通过调试,
unknown: 执行失败或解析脚本输出失败

    */
    string status = 1 ;
    /*
    执行日志
    */
    string msg = 2 ;
    /*
    指标组列表
    */
    repeated MetricGroups metric_groups = 3 ;
}

/*
DebugCollectorApi返回
*/
message DebugCollectorResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    collector.DebugCollectorResponse data = 4 ;
}

